x/cronos/keeper/keeper.go

```diff
@@ -215,3 +211,46 @@ func (k Keeper) IbcTransferCoins(ctx sdk.Context, from, destination string, coin
 	}()
 	return nil
 }
+
+// GetExternalContractByDenom find the corresponding contract for the denom,
+// external contract is taken in preference to auto-deployed one
+func (k Keeper) GetExternalContractByDenom(ctx sdk.Context, denom string) (common.Address, bool) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.DenomToExternalContractKey(denom))
+	if len(bz) == 0 {
+		return common.Address{}, false
+	}
+
+	return common.BytesToAddress(bz), true
+}
+
+// GetAutoContractByDenom find the corresponding auto-deployed contract for the denom,
+func (k Keeper) GetAutoContractByDenom(ctx sdk.Context, denom string) (common.Address, bool) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.DenomToAutoContractKey(denom))
+	if len(bz) == 0 {
+		return common.Address{}, false
+	}
+
+	return common.BytesToAddress(bz), true
+}
+
+// GetContractByDenom find the corresponding contract for the denom,
+// external contract is taken in preference to auto-deployed one
+func (k Keeper) GetContractByDenom(ctx sdk.Context, denom string) (contract common.Address, found bool) {
+	contract, found = k.GetExternalContractByDenom(ctx, denom)
```

Comment:
could `GetExternalContractByDenom` and `GetAutoContractByDenom` be private in that case?

Reply:
done